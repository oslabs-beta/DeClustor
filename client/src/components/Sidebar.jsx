import React, { useEffect, useState } from 'react'
import {
  Box,
  Divider,
  Drawer,
  IconButton,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  Typography,
  useTheme,
} from '@mui/material'
import {
  ChevronLeft,
  ChevronRightOutlined,
  HomeOutlined,
  CalendarMonthOutlined,
  AdminPanelSettingsOutlined,
  TrendingUpOutlined,
} from '@mui/icons-material'
import { useLocation, useNavigate } from 'react-router-dom'
import FlexBetween from './FlexBetween'
import profileImage from '../assests/profile.png'
import SsidChartOutlinedIcon from '@mui/icons-material/SsidChartOutlined'
import LanOutlinedIcon from '@mui/icons-material/LanOutlined'
import AssignmentOutlinedIcon from '@mui/icons-material/AssignmentOutlined'
import logo from '../assests/logo.png' 


{
  /* <Sidebar
isNonMobile={isNonMobile}
drawerWidth='250px'
isSidebarOpen={isSidebarOpen}
setIsSideBarOpen={setIsSideBarOpen}
/> */
}
// passin props from Layout
const Sidebar = ({
  isNonMobile,
  drawerWidth,
  isSidebarOpen,
  setIsSidebarOpen,
  // user
}) => {
  // grap the path that we currently at
  const { pathname } = useLocation()
  // state of currently page or track of which page is active right now
  const [active, setActive] = useState('')
  const navigate = useNavigate()
  // from theme color
  const theme = useTheme()

  // everytime path name has changed , set the active to the current page
  useEffect(() => {
    // set to currect url and determain which page we are on
    setActive(pathname.substring(1))
  }, [pathname])

  const navItems = [
    {
      text: 'Dashboard',
      icon: <HomeOutlined />,
    },
    {
      text: 'Task Overview',
      icon: <LanOutlinedIcon />,
    },
    {
      text: 'Cluster Metics',
      icon: <SsidChartOutlinedIcon />,
    },
    {
      text: 'Service',
      icon: <AssignmentOutlinedIcon />,
    },
    {
      text: 'Logs',
      icon: <CalendarMonthOutlined />,
    },
    {
      text: 'Management',
      icon: null,
    },
    {
      text: 'Admin',
      icon: <AdminPanelSettingsOutlined />,
    },
    {
      text: 'Performance',
      icon: <TrendingUpOutlined />,
    },
  ]

  return (
    // react drawer from react dom
    // persistenr drawer
    <Box component="nav">
      {isSidebarOpen && (
        <Drawer
          open={isSidebarOpen}
          onClose={() => setIsSidebarOpen(false)}
          variant="persistent"
          anchor="left"
          sx={{
            width: drawerWidth,
            // class name ที่ MUI ใช้กำหนดสำหรับส่วนของ Drawer component ที่ประพฤติเหมือนกระดาษ (paper), โดยปกติจะเป็นส่วนที่เลื่อนเข้าออกได้.
            '& .MuiDrawer-paper': {
              color: theme.palette.secondary[200],
              backgroundColor: theme.palette.background.alt,
              boxSizing: 'border-box',
              borderWidth: isNonMobile ? 0 : '2px',
            },
          }}
        >
          <Box width="230px">
            {/* t r b l */}
            <Box margin="1.2rem 1.8rem 0.8rem 1rem" >
              <FlexBetween color={theme.palette.secondary.main}>
              <Box
                component="img"
                alt="logo"
                src={logo}
                height="100px"
                width="100px"
                borderRadius="28%"
                sx={{
                  objectFit: 'cover',
                  borderColor: theme.palette.primary[400],
                  borderStyle: 'solid', 
                  borderWidth: 1,
                  marginLeft: '47px',
                  padding: '5px'
                }}
              />
                {/* <Box display="flex" alignItems="center" gap="0.5rem">
                  <Typography variant="h3" fontWeight="bold">
                    DeClustor
                  </Typography>
                </Box> */}

                {/* responsive for mobile , it's will pop up the left arrow */}
                {!isNonMobile && (
                  <IconButton onClick={() => setIsSidebarOpen(!isSidebarOpen)}>
                    <ChevronLeft />
                  </IconButton>
                )}
              </FlexBetween>
            </Box>
            {/* creating nav items // loop thru the navItems function
                check if icon is not existed 
                then set key to text */}
            <List>
              {navItems.map(({ text, icon }) => {
                if (!icon) {
                  return (
                    <Typography key={text} sx={{ m: '2.25rem 0 1rem 3rem' }}>
                      {text}
                    </Typography>
                  )
                }
                
                const lowerCaseText = text.toLowerCase()

                // set the navagat by following the {text} navItems name
                // swich the colors follow by if it's active?
                return (
                  <ListItem key={text} disablePadding>
                    <ListItemButton
                      onClick={() => {
                        navigate(`/${lowerCaseText}`)
                        setActive(lowerCaseText)
                      }}
                      sx={{
                        backgroundColor:
                          active === lowerCaseText
                            ? theme.palette.secondary[400]
                            : 'transparent',
                        color:
                          active === lowerCaseText
                            ? theme.palette.primary[600]
                            : theme.palette.secondary[100],
                      }}
                    >
                      <ListItemIcon
                        sx={{
                          ml: '0.8rem',
                          color:
                            active === lowerCaseText
                              ? theme.palette.primary[600]
                              : theme.palette.secondary[200],
                        }}
                      >
                        {icon}
                      </ListItemIcon>

                      <ListItemText primary={text} />
                      {/* arrow right */}
                      {active === lowerCaseText && (
                        <ChevronRightOutlined sx={{ ml: 'auto' }} />
                      )}
                    </ListItemButton>
                  </ListItem>
                )
              })}
            </List>
          </Box>
        
          {/* user profile need an update!*/}
          <Divider sx={{ width: '100%', maxWidth: '500px' , marginTop: '230px' }} />
          <Box position="absolute" bottom="2rem">
            <FlexBetween textTransform="none" gap="1rem" m="1.5rem 2rem 0 3rem">
              <Box
                component="img"
                alt="profile"
                src={profileImage}
                height="60px"
                width="60px"
                borderRadius="50%"
                sx={{ objectFit: 'cover' }}
              />
              <Box textAlign="left">
                <Typography
                  fontWeight="bold"
                  fontSize="0.9rem"
                  sx={{ color: theme.palette.secondary[100] }}
                >
                  {/* change to user info !! call the api '/userProfile' ?*/}
                  Ploy
                </Typography>
              </Box>
              {/* <SettingsOutlined
                sx={{
                  color: theme.palette.secondary[300],
                  fontSize: "25px ",
                }}
              /> */}
            </FlexBetween>
          </Box>
        </Drawer>
      )}
      ;
    </Box>
  )
}

export default Sidebar
