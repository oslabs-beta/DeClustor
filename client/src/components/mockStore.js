import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; // Import the mock store
import { createStore } from 'redux';
import rootReducer from '../redux/rootReducer'; // Adjust the path

const mockStore = configureStore([]);

export const renderWithRouterAndRedux = (
  ui,
  { route = '/', initialState = {} } = {}
) => {
  window.history.pushState({}, 'Test page', route);

  const store = mockStore(initialState); // Create a mock store with initial state

  return render(
    <Provider store={store}>
      <Router>{ui}</Router>
    </Provider>
  );
};
