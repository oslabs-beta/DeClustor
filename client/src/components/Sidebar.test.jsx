import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import { BrowserRouter as Router } from 'react-router-dom';
import Sidebar from './Sidebar';

import { renderWithRouterAndRedux } from './mockStore.js';
// Mock theme
const theme = createTheme({
  palette: {
    secondary: {
      200: '#9e9e9e',
      100: '#f5f5f5',
      300: '#e0e0e0',
      400: '#bdbdbd',
      main: '#ff4081',
    },
    background: {
      alt: '#f5f5f5',
    },
    main: {
      400: '#42a5f5',
      600: '#1e88e5',
    },
    mode: 'light', // Ensure mode is set
  },
  transitions: {
    duration: {
      enteringScreen: 225,
      leavingScreen: 195,
    },
  },
});

const mockStore = configureStore([]);

const renderWithRouter = (ui, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: Router });
};

describe('Sidebar Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      user: {
        username: 'John Doe',
      },
    });
  });

  it('renders correctly and matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router>
            <Sidebar
              isNonMobile={true}
              drawerWidth={240}
              isSidebarOpen={true}
              setIsSidebarOpen={jest.fn()}
            />
          </Router>
        </ThemeProvider>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('displays the correct user information', () => {
    const { getByText } = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router>
            <Sidebar
              isNonMobile={true}
              drawerWidth={240}
              isSidebarOpen={true}
              setIsSidebarOpen={jest.fn()}
            />
          </Router>
        </ThemeProvider>
      </Provider>
    );
    expect(getByText('John Doe')).toBeInTheDocument();
  });

  it('navigates correctly when a nav item is clicked', () => {
    const { getByText } = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router>
            <Sidebar
              isNonMobile={true}
              drawerWidth={240}
              isSidebarOpen={true}
              setIsSidebarOpen={jest.fn()}
            />
          </Router>
        </ThemeProvider>
      </Provider>
    );

    const dashboardLink = getByText('Dashboard');
    fireEvent.click(dashboardLink);
    expect(dashboardLink).toHaveStyle(`color: ${theme.palette.primary[600]}`);
  });

  // test('toggles sidebar when the close button is clicked', () => {
  //   const setIsSidebarOpenMock = jest.fn();
  //   const initialState = {
  //     user: {
  //       /* Mock user data */
  //     },
  //   };

  //   const { getByTestId } = renderWithRouterAndRedux(
  //     <Sidebar isOpen={true} onClose={setIsSidebarOpenMock} />,
  //     { initialState }
  //   );

  //   const closeButton = getByTestId('close-sidebar-button');
  //   fireEvent.click(closeButton);

  //   expect(setIsSidebarOpenMock).toHaveBeenCalledWith(false);
  // });
});
