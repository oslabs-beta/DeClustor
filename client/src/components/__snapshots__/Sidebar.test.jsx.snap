// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component renders correctly and matches the snapshot 1`] = `
<div>
  <nav
    class="MuiBox-root css-0"
  >
    <div
      class="MuiDrawer-root MuiDrawer-docked css-1y32at3-MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
      >
        <div
          class="MuiBox-root css-1n9fze6"
        >
          <div
            class="MuiBox-root css-n8mevj"
          >
            <div
              class="MuiBox-root css-18n9tnh"
            >
              <img
                alt="logo"
                class="MuiBox-root css-10kwszz"
                src="test-file-stub"
              />
            </div>
          </div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1mfzeo7-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-4dc74k-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="HomeOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m12 5.69 5 4.5V18h-2v-6H9v6H7v-7.81zM12 3 2 12h3v8h6v-6h2v6h6v-8h3z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Dashboard
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1mfzeo7-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-4dc74k-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="LanOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 22h8v-7h-3v-4h-5V9h3V2H8v7h3v2H6v4H3v7h8v-7H8v-2h8v2h-3zM10 7V4h4v3zM9 17v3H5v-3zm10 0v3h-4v-3z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Task Overview
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1mfzeo7-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-4dc74k-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="SsidChartOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M21 5.47 12 12 7.62 7.62 3 11V8.52L7.83 5l4.38 4.38L21 3zM21 15h-4.7l-4.17 3.34L6 12.41l-3 2.13V17l2.8-2 6.2 6 5-4h4z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Cluster Metics
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1mfzeo7-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-4dc74k-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AssignmentOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 15h7v2H7zm0-4h10v2H7zm0-4h10v2H7zm12-4h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-.14 0-.27.01-.4.04-.39.08-.74.28-1.01.55-.18.18-.33.4-.43.64-.1.23-.16.49-.16.77v14c0 .27.06.54.16.78s.25.45.43.64c.27.27.62.47 1.01.55.13.02.26.03.4.03h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m-7-.25c.41 0 .75.34.75.75s-.34.75-.75.75-.75-.34-.75-.75.34-.75.75-.75M19 19H5V5h14z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Service
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1mfzeo7-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-4dc74k-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CalendarMonthOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 16H5V10h14zm0-12H5V6h14zM9 14H7v-2h2zm4 0h-2v-2h2zm4 0h-2v-2h2zm-8 4H7v-2h2zm4 0h-2v-2h2zm4 0h-2v-2h2z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Logs
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <p
              class="MuiTypography-root MuiTypography-body1 css-13i4hj5-MuiTypography-root"
            >
              Account Management
            </p>
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1mfzeo7-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-4dc74k-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AdminPanelSettingsOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill-rule="evenodd"
                    >
                      <circle
                        cx="17"
                        cy="15.5"
                        r="1.12"
                      />
                      <path
                        d="M17 17.5c-.73 0-2.19.36-2.24 1.08.5.71 1.32 1.17 2.24 1.17s1.74-.46 2.24-1.17c-.05-.72-1.51-1.08-2.24-1.08"
                      />
                      <path
                        d="M18 11.09V6.27L10.5 3 3 6.27v4.91c0 4.54 3.2 8.79 7.5 9.82.55-.13 1.08-.32 1.6-.55C13.18 21.99 14.97 23 17 23c3.31 0 6-2.69 6-6 0-2.97-2.16-5.43-5-5.91M11 17c0 .56.08 1.11.23 1.62-.24.11-.48.22-.73.3-3.17-1-5.5-4.24-5.5-7.74v-3.6l5.5-2.4 5.5 2.4v3.51c-2.84.48-5 2.94-5 5.91m6 4c-2.21 0-4-1.79-4-4s1.79-4 4-4 4 1.79 4 4-1.79 4-4 4"
                      />
                    </g>
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Setting
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
            >
              <div
                class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1mfzeo7-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-4dc74k-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="TrendingUpOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m16 6 2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Performance
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </li>
          </ul>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-3p56ps-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-yyi0g8"
        >
          <div
            class="MuiBox-root css-11y2fid"
          >
            <img
              alt="profile"
              class="MuiBox-root css-13yu7l2"
              src="test-file-stub"
            />
            <div
              class="MuiBox-root css-13brihr"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1eqs37h-MuiTypography-root"
              >
                John Doe
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    ;
  </nav>
</div>
`;
